src/Extractor/BaseExtractor.php

```diff
@@ -385,19 +314,8 @@ protected function getDbParameters(): array
         return $this->dbParameters;
     }
 
-    protected function canFetchMaxIncrementalValueSeparately(bool $isAdvancedQuery): bool
-    {
-        return !$isAdvancedQuery && isset($this->incrementalFetching) && !$this->hasIncrementalLimit();
-    }
-
-    protected function hasIncrementalLimit(): bool
+    protected function canFetchMaxIncrementalValueSeparately(ExportConfig $export): bool
     {
-        if (!$this->incrementalFetching) {
-            return false;
-        }
-        if (isset($this->incrementalFetching['limit']) && (int) $this->incrementalFetching['limit'] > 0) {
-            return true;
-        }
-        return false;
+        return !$export->hasQuery() && $export->isIncremental() && !$export->hasIncrementalLimit();
```

Comment:
`!$isAdvancedQuery` => `!$export->hasQuery()`

`isset($this->incrementalFetching)` => `$export->isIncremental()`

`!$this->hasIncrementalLimit()` => `!$export->hasIncrementalLimit()`